Crash: enumeration constant with same name as existing identifier + non-`int` type

```c
int X = 0;
enum E {
    X = 4294967295
};
```

```
panic: reached unreachable code
/Users/ehaas/local/zig-macos-x86_64-0.12.0-dev.2619+5cf138e51/lib/std/debug.zig:403:14: 0x1074ff9bc in assert (arocc)
    if (!ok) unreachable; // assertion failure
             ^
/Users/ehaas/source/arocc/src/aro/Value.zig:35:21: 0x1075b46b9 in ref (arocc)
    std.debug.assert(v.opt_ref != .none);
                    ^
/Users/ehaas/source/arocc/src/aro/Value.zig:668:46: 0x10771c590 in compare (arocc)
    const lhs_key = comp.interner.get(lhs.ref());
                                             ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:5917:31: 0x1077ffbe9 in intFitsInType (arocc)
        return res.val.compare(.lte, max_int, p.comp) and
                              ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:2574:43: 0x1077af7d4 in enumSpec (arocc)
            else if (try res.intFitsInType(p, Type.int))
                                          ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:2020:47: 0x107725b4a in typeSpec (arocc)
                const enum_ty = try p.enumSpec();
```

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -2567,6 +2567,7 @@ fn enumSpec(p: *Parser) Error!Type {
             if (field.ty.eql(Type.int, p.comp, false)) continue;
 
             const sym = p.syms.get(field.name, .vars) orelse continue;
+            if (sym.kind != .enumeration) continue; // already an error
 
             var res = Result{ .node = field.node, .ty = field.ty, .val = sym.val };
             const dest_ty = if (p.comp.fixedEnumTagSpecifier()) |some|

--- a/test/cases/redefinitions.c
+++ b/test/cases/redefinitions.c
@@ -91,6 +91,11 @@ void f6(void) {}
 void f7(void);
 void f7() {}
 
+int X;
+enum E {
+    X = 4294967295,
+};
+
 #define EXPECTED_ERRORS "redefinitions.c:4:5: error: redefinition of 'foo' as different kind of symbol" \
     "redefinitions.c:1:5: note: previous definition is here" \
     "redefinitions.c:5:5: error: redefinition of 'foo' as different kind of symbol" \
@@ -130,4 +135,6 @@ void f7() {}
     "redefinitions.c:82:5: note: previous definition is here" \
     "redefinitions.c:86:5: error: redefinition of 'f5' with a different type" \
     "redefinitions.c:85:5: note: previous definition is here" \
+    "redefinitions.c:96:5: error: redefinition of 'X' as different kind of symbol" \
+    "redefinitions.c:94:5: note: previous definition is here" \
 

